cpp/include/cudf/utilities/span.hpp

```diff
@@ -198,7 +198,7 @@ struct device_span : public span_base<T, Extent, device_span<T, Extent>> {
  *
  * The index operator returns the corresponding row as a span.
  */
-template <typename T>
+template <typename T, typename Row>
```

Comment:
Is there anything that guarantees the `value_type` of `Row` is the same as `T`? In my earlier comment I made it a template-template parameter to ensure that they would always be the same.

Reply:
Yup, there was no guarantee that the types would match. 
I switched to a template template parameter, but had to include Extent in the template parameter list. 
This made the SFINAEd constructor a bit verbose, could not find a way to avoid this.
```  
  template <typename OtherT,
            template <typename, size_t>
            typename OtherRowType,
            typename std::enable_if<std::is_convertible<OtherRowType<OtherT, dynamic_extent>,
                                                        RowType<T, dynamic_extent>>::value,
                                    void>::type* = nullptr>
  constexpr base_2dspan(base_2dspan<OtherT, OtherRowType> const& other) noexcept
    : _data{other.data()}, _size{other.size()}
  {
  }